mmv1/third_party/terraform/functions/region_from_zone.go

```diff
@@ -41,11 +43,23 @@ func (f RegionFromZoneFunction) Run(ctx context.Context, req function.RunRequest
 		return
 	}
 
-	// Validate input - can't be an empty string, every zone will have 2nd to last element in string be a `-`
-	if len(arg0) != 0 && arg0[len(arg0)-2] == '-' {
-		resp.Diagnostics.Append(resp.Result.Set(ctx, arg0[:len(arg0)-2])...)
-	} else {
+	if arg0 == "" {
+		resp.Diagnostics.AddArgumentError(
+			0,
+			noMatchesErrorSummary,
+			"The input string is empty.",
```

Comment:
This might be me being extra, but the feedback you're giving to the user here is implying the problem but not stating it 100% directly. Could you please update to something like:

```suggestion
			"The input string cannot be empty.",
```